src/lib/js/array.rs

```diff
@@ -435,6 +435,46 @@ pub fn find(this: &Value, args: &[Value], interpreter: &mut Interpreter) -> Resu
     Ok(Gc::new(ValueData::Undefined))
 }
 
+/// Array.prototype.slice ( [begin[, end]] )
+///
+/// The slice method takes two arguments, start and end, and returns an array containing the
+/// elements of the array from element start up to, but not including, element end (or through the
+/// end of the array if end is undefined). If start is negative, it is treated as length + start
+/// where length is the length of the array. If end is negative, it is treated as length + end where
+/// length is the length of the array.
+/// <https://tc39.es/ecma262/#sec-array.prototype.slice>
+pub fn slice(this: &Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let new_array = make_array(&to_value(Object::default()), &[], interpreter)?;
+    let len: i32 =
+        from_value(this.get_field_slice("length")).expect("Could not convert argument to i32");
+
+    let start = match args.get(0) {
+        Some(v) => from_value(v.clone()).expect("failed to parse argument for Array method"),
+        None => return Ok(this.clone()),
```

Comment:
Should be returning the new array here. `this.clone()` will only copy the reference. Test case:
```
var animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
var animals2 = animals.splice();
animals.push(5);
console.log(animals, animals2);
> Array ["ant", "bison", "camel", "duck", "elephant", 5] Array []
```